contract.md

```diff
@@ -12,29 +16,76 @@ like to work on, as part of your project (for example, other things related to D
+
+I have initially chosen to work with Scala because I want to be more familiar with the language and leverage the fact that it is very suitable for creating DSLs. I will also leverage many libraries that were initially built on Java, but should work in Scala. Nonetheless, I have never tried hooking libraries up across language before, so this will be a learning point for me in this project.
+
 ## Time management plan
 
 _How do you plan to set aside time outside of class, to work on the project? Are there
 intermediate milestones that you can create, to help you make consistent progress?_
 
+I plan to set aside 6-8 hours per week outside of class to work on the project. I will be spreading my time so that I spent at least 1 hour before Wednesday studio sessions so that I can get help quickly on any initial questions I have for the tasks that week. I also plan to write clear documentation as I go in my GitHub repo, as a Wiki page or through markdown docs. This would help me and my review group clearly see my progress within this project.
+
+Here is a timeline that I plan to follow with intermediate milestones as indicated in bold.
+
+Mar 26 | project plan and contract (this github)
+    
+Apr 2 |
+- **create first draft of ideal dsl syntax and AST**
+- deeper research into libraries for operations to parse documents in Scala and Java
+- **set up project and test functions**: writing the necessary operations in GPL with the help of imported libraries, ex. I/O: open and read file in (format) + generate new file in same format
+- design start symbol, production rules, terminals from ideal syntax to formalize the DSL
+
+Apr 9 |
+- **implement parsers to generate AST for basic workflow**
+-	edit ideal syntax to balance with capabilities of Scala
+-	implement functions needed for other workflows
+- stretch goal for week: complete “base level” requirements for dsl and project - takes a file written in my DSL with keywords and instructions for organization, another word file as doc to organize, and outputs a file as structured based on dsl commands.
+
+Apr 16 | 
+- **Complete base level requirements for dsl/project, and revise documentation in and outside of code for ease of use.**
+
+Apr 23 | 
+- **Complete low-difficulty extensions to project** - ex. handling other documentation formats
+- (Stretch goal) Complete other extensions to project depending on time. I would like to create a CLI or GUI of some sort!
+
+Apr 28 | Complete any started extensions to the best of my capabilities!
+
```

Comment:
I think your roadmap is very doable over the next few weeks. All of the detail that you added should make it easy for you to track your progress